src/components/InstantSearch.js

```diff
@@ -0,0 +1,91 @@
+import instantsearch from 'instantsearch.js/es/';
+import { createInstantSearchComponent } from '../util/createInstantSearchComponent';
+import { warn } from '../util/warn';
+
+const oldApi = () =>
+  warn(
+    `Vue InstantSearch: You used the prop api-key or api-key.
+These have been replaced by search-client.
+
+See more info here: https://community.algolia.com/vue-instantsearch/components/InstantSearch.html#usage`
+  );
+
+export default createInstantSearchComponent({
+  name: 'AisInstantSearch',
+  props: {
+    searchClient: {
+      type: Object,
+      required: true,
+    },
+    indexName: {
+      type: String,
+      required: true,
+    },
+    routing: {
+      default: null,
+      validator(value) {
+        if (
+          typeof value === 'boolean' ||
+          !value.router ||
+          !value.stateMapping
+        ) {
+          warn('routing should be an object, with `router` and `stateMapping`');
```

Comment:
We should link to the documentation: https://www.algolia.com/doc/api-reference/widgets/instantsearch/vue/#widget-param-routing

Also perhaps warn a more human-friendly message.